src/lib.rs

```diff
@@ -61,22 +62,29 @@ use iron::headers;
 /// The struct that holds the CORS configuration.
 pub struct CorsMiddleware {
     allowed_hosts: Option<HashSet<String>>,
+    allow_credentials: bool,
 }
 
 impl CorsMiddleware {
     /// Specify which origin hosts are allowed to access the resource.
     pub fn with_whitelist(allowed_hosts: HashSet<String>) -> Self {
         CorsMiddleware {
             allowed_hosts: Some(allowed_hosts),
+            allow_credentials: false,
         }
     }
 
+    pub fn allow_credentials(&mut self) {
+        self.allow_credentials = true;
+    }
```

Comment:
I'm not sure yet about this API. I think a setter method like this is not a good solution.

I could imagine two ways to configure the credentials: Either as constructor argument, or as a builder. Maybe something like this?

```rust
CorsMiddlewareBuilder::with_whitelist(allowed_hosts) // Or: with_allow_any()
    .allow_credentials(true)
    .build();
```

What are your thoughts?

Reply:
I agree
I'm going to change tonight.

But this way the constract will change.

Is this a problem for you?

Reply:
We could deprecate the constructor functions (`with_whitelist` and `allow_credentials`) on the `CorsMiddleware` object itself and simply add the builder in parallel. This way it's compatible and we can remove it in the next version.